src/v2/api/services/ScheduleService.ispec.ts

```diff
@@ -556,9 +589,231 @@ describe('ScheduleService', () => {
+    });
+  });
+
+  describe('deleteEvent', () => {
+    it('should delete an event without associated lessons', async () => {
+      jest.useFakeTimers().setSystemTime(new Date('2022-09-05T00:01:00'));
+      const eventToDelete = {
+        id: 'some-event-1st-semester-no-period-09-12',
+        name: 'name4',
+        groupId: 'group',
+        period: Period.NO_PERIOD,
+        group: {
+          code: 'AA-12',
+          createdAt: expect.any(Date),
+          id: 'group',
+          updatedAt: expect.any(Date),
+        },
+        eventInfo: [
+          {
+            createdAt: expect.any(Date),
+            description: 'something',
+            eventId: 'some-event-1st-semester-no-period-09-12',
+            number: 0,
+            updatedAt: expect.any(Date),
+          },
+        ],
+        startTime: new Date('2022-09-12T08:30:00'),
+        endTime: new Date('2022-09-12T10:00:00'),
+        createdAt: expect.any(Date),
+        updatedAt: expect.any(Date),
+        lessons: [],
+        url: null,
+      };
+
+      const deletedEvent = await scheduleService.deleteEvent(eventToDelete.id);
+      expect(deletedEvent.event).toEqual(eventToDelete);
+    });
+
+    it('should delete an event with associated lessons and discipline', async () => {
+      const deletedEventID = 'anotherGroup-selective-practice-event-1st-semester-every-week-09-12';
+      jest.useFakeTimers().setSystemTime(new Date('2022-09-05T00:01:00'));
+      const eventToDelete = {
+        id: 'anotherGroup-selective-practice-event-1st-semester-every-week-09-12',
+        name: 'name10',
+        groupId: 'anotherGroup',
+        period: Period.EVERY_WEEK,
+        group: {
+          code: 'AA-11',
+          createdAt: expect.any(Date),
+          id: 'anotherGroup',
+          updatedAt: expect.any(Date),
+        },
+        startTime: new Date('2022-09-12T08:30:00'),
+        endTime: new Date('2022-10-03T10:00:00'),
+        createdAt: expect.any(Date),
+        updatedAt: expect.any(Date),
+        lessons: [
+          {
+            createdAt: expect.any(Date),
+            disciplineType: {
+              createdAt: expect.any(Date),
+              disciplineId: 'anotherGroup-discipline',
+              id: 'anotherGroup-selectiveDiscipline-practice',
+              name: 'PRACTICE',
+              updatedAt: expect.any(Date),
+            },
+            disciplineTypeId: 'anotherGroup-selectiveDiscipline-practice',
+            eventId: 'anotherGroup-selective-practice-event-1st-semester-every-week-09-12',
+            updatedAt: expect.any(Date),
+          },
+        ],
+        url: null,
+      };
+
+      const { event: { eventInfo, ...event } } = await scheduleService.deleteEvent(deletedEventID);
+
+      expect(eventInfo.length).toBe(3);
+      expect(event).toStrictEqual(eventToDelete);
+    });
+
+    it('should delete disciplineTeachers and discipline', async () => {
```

Comment:
The previous test also deletes teachers and discipline types, you should also test the event deletion if there are still relevant events for the discipline type